crates/adapterlib/src/transport.rs

```diff
@@ -62,6 +62,38 @@ pub trait IntegratedInputEndpoint: InputEndpoint {
 
 /// Commands for an [InputReader] to execute.
 ///
+/// # Transitions
+///
+/// The following diagram shows the possible order in which the controller can
+/// issue commands to [InputReader]s:
+///
+/// ```
+///   ┌─⯇─ (start) ─⯈──┐
```

Comment:
this is complicated...
I can't check that it's correct, but the FSM looks reasonable to me